src/trunk.c

```diff
@@ -8811,75 +8824,135 @@ trunk_branch_count_num_tuples(trunk_handle *spl,
+
+   size_t nkvbytes = 0;
    for (uint16 branch_no = start_branch; branch_no != end_branch;
         branch_no        = trunk_add_branch_number(spl, branch_no, 1))
    {
       uint64 addr = trunk_get_branch(spl, &node, branch_no)->root_addr;
       uint64 num_tuples_in_branch;
-      uint64 kv_bytes_in_branch;
+      uint64 kv_bytes_in_branch = 0;
+      num_branches++;
+
       trunk_branch_count_num_tuples(
          spl, &node, branch_no, &num_tuples_in_branch, &kv_bytes_in_branch);
-      uint64 kib_in_branch = 0;
+
+      nkvbytes = kv_bytes_in_branch;
+
+      sum_ntuples += num_tuples_in_branch;
+      sum_nkvbytes += kv_bytes_in_branch;
+
+      uint64 kib_in_branch = 1;
       // trunk_branch_extent_count(spl, &node, branch_no);
       kib_in_branch *= B_TO_KiB(trunk_extent_size(&spl->cfg));
       fraction space_amp =
-         init_fraction(kib_in_branch * 1024, kv_bytes_in_branch);
+         init_fraction(KiB_TO_B(kib_in_branch), kv_bytes_in_branch);
+
       platform_log(
          log_handle,
-         "| %6u | %12lu | %12lu | %9luKiB | %8luKiB |   " FRACTION_FMT(
+         "    | %6u | %12lu | %12lu |%12lu %14s| %8lu KiB|   " FRACTION_FMT(
             2, 2) "   |\n",
          branch_no,
          addr,
          num_tuples_in_branch,
-         B_TO_KiB(kv_bytes_in_branch),
+         nkvbytes,
+         size_fmtstr("(%s)", nkvbytes),
          kib_in_branch,
          FRACTION_ARGS(space_amp));
    }
-   platform_log(
-      log_handle,
-      "------------------------------------------------------------------\n");
-   platform_log(log_handle, "\n");
    trunk_node_unget(spl->cc, &node);
+
+   platform_log(log_handle, "%s\n", dashes_long);
+   platform_log(log_handle,
+                "Sum | %6u | %12s | %12lu |%12lu %14s| %8s    |   \n",
+                num_branches,
+                "",
+                sum_ntuples,
+                sum_nkvbytes,
+                size_fmtstr("(%s)", sum_nkvbytes),
+                "");
+
+   platform_log(log_handle, "%s\n", dashes_long);
+   platform_log(log_handle, "\n");
+
+   // Aggregate metrics
+   node_stats->num_nodes++;
+   node_stats->num_tuples += sum_ntuples;
+   node_stats->num_kv_bytes += sum_nkvbytes;
+
    return TRUE;
 }
 
+/* Driver function to print summary branch information for all trunk nodes. */
 void
 trunk_print_branches(platform_log_handle *log_handle, trunk_handle *spl)
 {
-   trunk_for_each_node(spl, trunk_node_print_branches, log_handle);
+   trunk_node_stats node_stats;
+   ZERO_STRUCT(node_stats);
+   node_stats.log_handle = log_handle;
+
+   platform_log(log_handle, "{\n");
+   trunk_for_each_node(spl, trunk_node_print_branches, (void *)&node_stats);
+
+   platform_log(log_handle,
+                "Number of trunk nodes=%lu, Num tuples=%lu"
+                ", Num KV-bytes=%lu (%s)\n",
+                node_stats.num_nodes,
+                node_stats.num_tuples,
+                node_stats.num_kv_bytes,
+                size_str(node_stats.num_kv_bytes));
```

Comment:
Adding a one-line summary of trunk-node metrics as gathered during print traversal.